### ISSUE TITLE
Errors in MGI GO annotation file from PAINT

### ISSUE THREAD

**Issue by pengyubio** [geneontology/helpdesk#155]
I see the following duplicated files. Is there anything wrong?

```
$ wget http://geneontology.org/gene-associations/gene_association.mgi.gz
$ zcat gene_association.mgi.gz | cut -f 2,3 | grep -w 'Gm4881\|Gm5890\|LOC382133\|Olfr485\|Olfr912\|Prn\|Rnf212b\|Slc12a6\|Sly\|Usp17le\|Zfp973' | sort -k 2,2 -k 1,1 | uniq
A0A140LIR9    Gm4881
MGI:3642958    Gm4881
A0A140LI04    Gm5890
MGI:3649012    Gm5890
MGI:5434094    LOC382133
MGI:5434172    LOC382133
A0A140LJF5    Olfr485
MGI:3030319    Olfr485
A0A1L1SSS5    Olfr912
MGI:3030746    Olfr912
MGI:5753213    Prn
MGI:97767    Prn
MGI:3642276    Rnf212b
MGI:5589964    Rnf212b
MGI:2135960    Slc12a6
MGI:5439454    Slc12a6
MGI:2687328    Sly
MGI:5434982    Sly
MGI:107697    Usp17le
MGI:3643640    Usp17le
MGI:3615331    Zfp973
MGI:3779822    Zfp973
```

**Comment by suzialeksander** [comment]
Hi @pengyubio , looking at MGI's site it looks like some of these may be secondary IDs (http://www.informatics.jax.org/marker/MGI:107697 has "MGI:3643640" as an "Other Accession"). However, in other cases these appear to be the incorrect accessions ( I checked MGI:5434982/Sly and MGI:3779822/Zfp973) so I'm hoping @ukemi can look into this.  

also mentioning @pgaudet and @huaiyumi in case this is a PAINT thing, all the ones I looked at were IBA.

**Comment by cmungall** [comment]
You're correct @suzialeksander. Huaiyu is on holiday. @dustine32 + @dougli1sqrd will look at this. cc @kltm

**Comment by dustine32** [comment]
Just checked MGI:5434982 (UniProtKB:Q810R0) and the name changed from Sly to Gm21627 around 12-20-17, after our most recent library update:
https://www.uniprot.org/uniprot/Q810R0?diff=true&version=88&version=92

So we wouldn't have updated the gene name yet. I'll check the MGI:3779822/Zfp973 one too.

**Comment by dustine32** [comment]
So MGI:3779822/Zfp973 is a little different. Panther maps it to UniProtKB:A2BE20, which has two MGI xrefs to MGI:3779822 and MGI:3615331. While the label on the MGI:3779822 xref is Zfp965, I guess we went with using Zfp973, the label of UniProtKB:A2BE20.
https://www.uniprot.org/uniprot/A2BE20

From looking at these examples, the errors appear to be related to nuances of UniProt's changes and cross-references.

**Comment by pgaudet** [comment]
Does @cmungall know what's going on?

Le ven. 31 août 2018 à 3:36 PM, dustine32 <notifications@github.com> a
écrit :

**Comment by suzialeksander** [comment]
@dougli1sqrd @dustine32 I just ran the wget & cat (as pasted above) this morning and have the exact same list as @pengyubio originally posted.  Is there anything you can add, or is this a question for @ukemi ?

**Comment by ukemi** [comment]
I think this is a question for the PAINT team. I just checked some of these and they are not in our source gaf. 
Here is what I find using these identifiers to search MGI:

Input	Input Type	MGI Gene/Marker ID	Symbol	Name	Feature Type
A0A140LIR9	TrEMBL	MGI:3642958	Gm4881	predicted gene 4881	protein coding gene
MGI:3642958	MGI	MGI:3642958	Gm4881	predicted gene 4881	protein coding gene
A0A140LI04	TrEMBL	MGI:3649012	Gm5890	predicted gene 5890	protein coding gene
MGI:3649012	MGI	MGI:3649012	Gm5890	predicted gene 5890	protein coding gene
MGI:5434094	MGI	MGI:5434094	Gm20738	predicted gene, 20738	protein coding gene
MGI:5434172	MGI	MGI:5434172	Gm20816	predicted gene, 20816	protein coding gene
A0A140LJF5	TrEMBL	MGI:3030319	Olfr485	olfactory receptor 485	protein coding gene
MGI:3030319	MGI	MGI:3030319	Olfr485	olfactory receptor 485	protein coding gene
A0A1L1SSS5	TrEMBL	MGI:3030746	Olfr912	olfactory receptor 912	protein coding gene
MGI:3030746	MGI	MGI:3030746	Olfr912	olfactory receptor 912	protein coding gene
MGI:5753213	MGI	MGI:97767	Prn	prion protein readthrough transcript	protein coding gene
MGI:97767	MGI	MGI:97767	Prn	prion protein readthrough transcript	protein coding gene
MGI:3642276		No associated gene			
MGI:5589964	MGI	MGI:5589964	Rnf212b	ring finger protein 212B	protein coding gene
MGI:2135960	MGI	MGI:2135960	Slc12a6	solute carrier family 12, member 6	protein coding gene
MGI:5439454	MGI	MGI:5439454	Gm21985	predicted gene 21985	protein coding gene
MGI:2687328	MGI	MGI:2687328	Sly	Sycp3 like Y-linked	protein coding gene
MGI:5434982	MGI	MGI:5434982	Gm21627	predicted gene, 21627	protein coding gene
MGI:107697	MGI	MGI:107697	Usp17le	ubiquitin specific peptidase 17-like E	protein coding gene
MGI:3643640	MGI	MGI:107697	Usp17le	ubiquitin specific peptidase 17-like E	protein coding gene
MGI:3615331	MGI	MGI:3615331	Zfp973	zinc finger protein 973	protein coding gene
MGI:3779822	MGI	MGI:3779822	Zfp965	zinc finger protein 965	protein coding gene

**Comment by pgaudet** [comment]
@huaiyumi @dustine32  Can you please have a look ?

**Comment by dustine32** [comment]
@pgaudet The PAINT mapping is based on the UniProt reference proteome entry when the Panther library is built. Most of these discrepancies are due to the UniProt entries being changed (e.g. gene symbol, xrefs) after the file date that the current library build (which PAINT is using) pulled as the source. For the data coming out of PAINT right now, that date is around April 2016.

We don't currently have a mechanism to more regularly update these mappings in Panther/PAINT but they will be updated when the new library is built before the end of this year.

**Comment by selewis** [comment]
@David Hill <David.Hill@jax.org> The GPI file won't work directly because
it's not merely mapping an MGI ID to a UniProt ID. Rather it has to
directly relate to the protein in the proteome file that is used to build
the PANTHER family trees each year.

Each MOD must send their GPI file (or the equivalent) to Maria/UniProt and
work with them to ensure that the next release of the GCRP (gene centric
reference proteomes) files are correct (both in terms of IDs and in terms
of the proteins). No short-circuiting the system in other words.

-S

**Comment by kltm** [comment]
[Originally from @selewis , removed email addresses]

@ukemi The GPI file won't work directly because
it's not merely mapping an MGI ID to a UniProt ID. Rather it has to
directly relate to the protein in the proteome file that is used to build
the PANTHER family trees each year.

Each MOD must send their GPI file (or the equivalent) to Maria/UniProt and
work with them to ensure that the next release of the GCRP (gene centric
reference proteomes) files are correct (both in terms of IDs and in terms
of the proteins). No short-circuiting the system in other words.
…

-S

On Tue, Nov 13, 2018 at 5:28 AM David Hill ***@***.***> wrote:
 Hi @dustine32 <https://github.com/dustine32>,

 Would it make sense to use the GPI file from MGI for the mapping? At least
 for the time when it was taken, it would be consistent with everything else.

 —
 You are receiving this because you are subscribed to this thread.
 Reply to this email directly, view it on GitHub
 <#155 (comment)>,
 or mute the thread

**Comment by suzialeksander** [comment]
@dustine32 

> they will be updated when the new library is built before the end of this year

Were these updated already? 

I don't see any of the duplications @pengyubio found, some secondary IDs are still being used.  Would like to close if this isn't an issue anymore.

**Comment by suzialeksander** [comment]
Also pinging @huaiyumi

**Comment by dustine32** [comment]
@suzialeksander Sorry, the new library has to yet been released to the point where the new changes would be reflected in the PAINT IBA files. We're still finishing up Panther version 14.1, then will update the PAINT DB with the new mappings. However, not all of the examples from above will be solved by this, possibly due to either Reference Proteome's mapping files or they way we currently parse them; still need to investigate.

These ID-to-symbol mappings should be fixed once 14.1 is released as they're simple data updates:
```
MGI:5434094    LOC382133
MGI:5434172    LOC382133
A0A1L1SSS5    Olfr912
MGI:3030746    Olfr912
MGI:5753213    Prn
MGI:97767    Prn
MGI:2135960    Slc12a6
MGI:5439454    Slc12a6
MGI:2687328    Sly
MGI:5434982    Sly
MGI:107697    Usp17le
MGI:3643640    Usp17le
```

**Comment by suzialeksander** [comment]
@dustine32 I just ran this search today and I'm seeing some of these errors that were going to be addressed (e.g. Sly).  Here are my results, note filenames have changed since ticket was opened:



<pre>
➜  wget http://geneontology.org/gene-associations/gene_association.mgi.gz

➜  zcat < /Users/suzi/mgi.gaf.gz | cut -f 2,3 | grep -w 'Gm4881\|Gm5890\|LOC382133\|Olfr485\|Olfr912\|Prn\|Rnf212b\|Slc12a6\|Sly\|Usp17le\|Zfp973' | sort -k 2,2 -k 1,1 | uniq


A0A140LIR9	Gm4881
A0A140LI04	Gm5890
MGI:5434094	LOC382133
MGI:5434172	LOC382133
A0A140LJF5	Olfr485
MGI:3030319	Olfr485
A0A1L1SSS5	Olfr912
MGI:3030746	Olfr912
MGI:5753213	Prn
MGI:97767	Prn
MGI:3642276	Rnf212b
MGI:5589964	Rnf212b
MGI:2135960	Slc12a6
MGI:5439454	Slc12a6
MGI:2687328	Sly
MGI:5434982	Sly
MGI:107697	Usp17le
MGI:3643640	Usp17le
MGI:3615331	Zfp973</pre>

Is there a different tracker where this may get more traffic? `panther-enrichment` doesn't seem quite right, or??

**Comment by dustine32** [comment]
Hi @suzialeksander , first off you can try moving this to https://github.com/pantherdb/Helpdesk .

On the mapping, this may sound ridiculous, but the [current GO release MGI file](http://current.geneontology.org/annotations/mgi.gaf.gz) still has PAINT annotations based on the older 13.1 library:
```
!Header copied from paint_mgi_valid.gaf
!=================================
!Created on Thu Jun  6 12:12:52 2019.
!PANTHER version: v.13.1.
```
If you check the `snapshot` [mgi.gaf.gz](http://snapshot.geneontology.org/annotations/mgi.gaf.gz), it shows the PANTHER version is 14.1 and your query results look a little better:
```
$ zcat < mgi.gaf.gz | cut -f 2,3 | grep -w 'Gm4881\|Gm5890\|LOC382133\|Olfr485\|Olfr912\|Prn\|Rnf212b\|Slc12a6\|Sly\|Usp17le\|Zfp973' | sort -k 2,2 -k 1,1 | uniq
A0A140LIR9	Gm4881
A0A140LI04	Gm5890
A0A140LJF5	Olfr485
MGI:3030319	Olfr485
MGI:3030746	Olfr912
MGI:97767	Prn
MGI:3642276	Rnf212b
MGI:5589964	Rnf212b
MGI:2135960	Slc12a6
MGI:2687328	Sly
MGI:107697	Usp17le
MGI:3615331	Zfp973
```
But not perfect. Still got the dup `Rnf212b` and `Olfr485` mappings. I'll have to look into why.

**Comment by suzialeksander** [comment]
Moved to https://github.com/pantherdb/Helpdesk/issues/11, this issue is upstream of GO.
